File: packages/zynth-markdown/ios/ZynthMarkdownModule.swift
import Foundation
import ZynthKit

@objc(ZynthMarkdownModule)
final class ZynthMarkdownModule: NSObject, ZynthModule, ZynthSyncModule {
  let name: String = "ZynthMarkdown"

  var exportedMethods: [String] {
    return ["parse"]
  }

  func call(method: String, args: ZynthArgs) throws -> Any? {
    switch method {
    case "parse":
      return ["result": parse(args)]
    default:
      throw ZynthModuleError.methodNotExported(module: name, method: method)
    }
  }

  func callSync(method: String, args: ZynthArgs) throws -> Any? {
    switch method {
    case "parse":
      return parse(args)
    default:
      throw ZynthModuleError.methodNotExported(module: name, method: method)
    }
  }

  private func parse(_ args: ZynthArgs) -> Any? {
    let content = args.string("content", default: "")
    let options = UInt32(args.number("options", default: 0))
    let extensions = UInt32(args.number("extensions", default: 0))
    let json = ZynthMarkdownBridge.parse(
      content,
      options: options,
      extensions: extensions
    )
    return json
  }
}

File: packages/zynth-markdown/android/ZynthMarkdown/src/main/java/dev/zynth/markdown/ZynthMarkdownModule.kt
package dev.zynth.markdown

import com.zynth.kit.runtime.ZynthModule
import com.zynth.kit.runtime.ZynthSyncModule
import com.zynth.kit.runtime.ZynthArgs
import org.json.JSONObject

class ZynthMarkdownModule : ZynthModule, ZynthSyncModule {
    override val name: String = "ZynthMarkdown"

    override val exportedMethods: List<String> = listOf("parse")

    override fun call(method: String, args: ZynthArgs): JSONObject {
        return when (method) {
            "parse" -> resultResponse(parse(args))
            else -> throw IllegalArgumentException("Unsupported method: $method")
        }
    }

    override fun callSync(method: String, args: ZynthArgs): Any {
        return when (method) {
            "parse" -> parse(args)
            else -> throw IllegalArgumentException("Unsupported method: $method")
        }
    }

    private fun parse(params: ZynthArgs): String {
        val content = params.getString("content", "")
        val options = params.getInt("options", 0)
        val extensions = params.getInt("extensions", 0)
        return ZynthMarkdownNative.parse(content, options, extensions) ?: ""
    }

    private fun resultResponse(result: Any?): JSONObject {
        return JSONObject().apply {
            put("result", result ?: JSONObject.NULL)
        }
    }
}
